drivers/stepper/step_dir/step_dir_stepper_common.h

```diff
@@ -27,19 +30,27 @@ struct step_dir_stepper_common_config {
 	const struct gpio_dt_spec step_pin;
 	const struct gpio_dt_spec dir_pin;
 	bool dual_edge;
+	const struct stepper_timing_source_api *timing_source;
+	const struct device *counter;
 };
 
 /**
  * @brief Initialize common step direction stepper config from devicetree instance.
+ *        If the counter property is set, the timing source will be set to the counter timing
+ * source.
  *
  * @param node_id The devicetree node identifier.
  */
 #define STEP_DIR_STEPPER_DT_COMMON_CONFIG_INIT(node_id)                                            \
 	{                                                                                          \
 		.step_pin = GPIO_DT_SPEC_GET(node_id, step_gpios),                                 \
-		.dir_pin = GPIO_DT_SPEC_GET(node_id, dir_gpios),	                           \
+		.dir_pin = GPIO_DT_SPEC_GET(node_id, dir_gpios),                                   \
 		.dual_edge = DT_PROP_OR(node_id, dual_edge_step, false),                           \
```

Comment:
sorry just caught this: booleans are implicitly "false" if not set, this can just be `DT_PROP(node_id, dual_edge_step)`, wanna change it while you are at it?

Reply:
Good call, changed it.